src/test/miner_tests.cpp

```diff
@@ -163,10 +167,12 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
     delete pblocktemplate;
     mempool.clear();
 
-    // orphan in mempool, template creation fails
+    // orphan in mempool not mined
     hash = tx.GetHash();
     mempool.addUnchecked(hash, entry.Fee(1000000).Time(GetTime()).FromTx(tx));
-    BOOST_CHECK_THROW(CreateNewBlock(chainparams, scriptPubKey), std::runtime_error);
+    BOOST_CHECK(pblocktemplate = CreateNewBlock(chainparams, scriptPubKey));
```

Comment:
Perhaps for consistency, add a check to ensure the transaction has been removed from the mempool.